service/controller/v14/resource/endpoint/current.go

```diff
@@ -46,27 +49,39 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		}
 	}
 
-	currentEndpoint := Endpoint{
-		IPs:              []string{},
-		ServiceName:      serviceName,
-		ServiceNamespace: pod.GetNamespace(),
-	}
-	k8sEndpoints, err := r.k8sClient.CoreV1().Endpoints(pod.GetNamespace()).Get(serviceName, metav1.GetOptions{})
-	if errors.IsNotFound(err) {
-		return nil, nil
-	} else if err != nil {
-		return nil, microerror.Mask(err)
-	}
+	var endpoint *Endpoint
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding endpoint")
+
+		endpoint = &Endpoint{
+			IPs:              []string{},
+			ServiceName:      serviceName,
+			ServiceNamespace: pod.GetNamespace(),
+		}
+
+		k8sEndpoints, err := r.k8sClient.CoreV1().Endpoints(pod.GetNamespace()).Get(serviceName, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			// In case the endpoint manifest cannot be found in the Kubernetes API we
+			// return the endpoint structure we dispatch without filling any IP.
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find endpoint")
```

Comment:
This is new. IMO we should always dispatch the structure the resource is concerned about. Only the information it carries change. This is for instance to have all information we need to do basic logging about the resource's processes.